### DISCUSSION TITLE
How to Use Font Awesome Extension

### DISCUSSION THREAD

**Discussion by eteitelbaum** [quarto-dev/quarto-cli#1619] [Q&A] [answered]
I'm trying to use the Font Awesome extension in Quarto in a website I am building. I installed the extension as instructed on the [Github page](https://github.com/quarto-ext/fontawesome) :

```
quarto install extension quarto-ext/fontawesome
```

After that, I can see the Font Awesome extension in the `_extension` file. 

Next, I try to insert an icon into `index.qmd` like this: 

`{{< fa thumbs-up >}}` 

and then render it. But nothing happens. 

Any ideas on how to troubleshoot? Is there another step? Something I have to add to `_quarto.yml`?

**Comment by mcanouil** [comment]
Could you provide a small reproducible example?
Did you try the example file?
https://github.com/quarto-ext/fontawesome/blob/main/example.qmd

**Reply by eteitelbaum** [reply]
Thanks. 

It seems to be a white space issue. If I align the font awesome icon with the margin it shows, but if I indent it, it doesn't. 

So this works: 

`{{< fa thumbs-up >}} `

But when I do this, nothing shows up: 

      `{{< fa thumbs-up >}} ` 

However, it is a little strange, because regardless of whether I write: 

`Good job! {{< fa thumbs-up >}} `

or: 

`Good job! `      `{{< fa thumbs-up >}} `

the icon shows up. So white space only matters when you are aligning the icon to the left margin?

**Reply by mcanouil** [reply]
Could you make a full example with the different cases?
For example:

```
---
title: "Untitled"
format: html
---

{{< fa thumbs-up >}}
```

It will be way easier with something like this to work on.
Here in your example you apparently put the shortcode between markdown inline code delimiters, s currently I do not fully understand in which cases in does not work.

**Reply by eteitelbaum** [reply]
Thanks. Quarto has now auto-corrected the code to eliminate the unhelpful white space. I am not sure why it wasn't doing that before, but everything is working now.  Probably not worth spending more time on at this point, but I'll go ahead and just paste my code as it it also gets auto-corrected. The first instance didn't work but the other three did.  The second one had a big space between "Good job!" and the icon code (which gets corrected by the GitHub code editor). 

`---
title: "Test"
---

This is a Quarto website.

To learn more about Quarto websites visit <https://quarto.org/docs/websites>.

```{r}
1 + 1
```

           {{< fa thumbs-up >}}

{{< fa thumbs-up >}}

Good job!       {{< fa thumbs-up >}}

Good job! {{< fa thumbs-up >}}`

**Reply by mcanouil** [reply]
I think this happens because for pandoc and markdown parsers, an indented text is not plain text which might confused the Lua filter bringing FA shortcodes.
It's kind of an expected behaviour, but I'll double check.

**Reply by mcanouil** [reply]
As you can see below, indented text is supposed to be code, thus it's not parsed properly in that context while `{{< meta title >}}` is (note: `{{< meta title >}}` in this context is formated as code, not as plain text as it would be without the indentation).
So I'll open an issue in https://github.com/quarto-ext/fontawesome to investigate this and see if there is something we can do and want to do.
<img width="1507" alt="screenshot of a web browser with on the left the rendered html containing several font awesome icons from shortcodes with one missing and on the right the developper console showing that the missing icon is a pre code html tags" src="https://user-images.githubusercontent.com/8896044/182245941-17f29314-d4cb-422f-8a45-9411b0ba3948.png">

**Reply by mcanouil** [reply, marked as answer]
See https://github.com/quarto-ext/fontawesome/issues/13
